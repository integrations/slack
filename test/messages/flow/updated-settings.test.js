const { UpdatedSettings } = require('../../../lib/messages/flow');
const { Subscription } = require('../../models');
const repository = require('../../fixtures/repo.json');

describe('UpdatedSettings message', () => {
  test('default settings', async () => {
    const subscription = new Subscription({
      channelId: 'C001',
    });

    const message = new UpdatedSettings({ subscription, resource: repository });
    expect(message.toJSON()).toMatchSnapshot();
  });

  test('sub settings', async () => {
    const subscription = new Subscription({
      channelId: 'C001',
      settings: { commits: 'all' },
    });

    const message = new UpdatedSettings({ subscription, resource: repository });
    expect(message.toJSON().attachments[0].text).toMatch(/commits:all/);
    expect(message.toJSON()).toMatchSnapshot();
  });

  test('hides overwritten settings', async () => {
    const subscription = new Subscription({
      channelId: 'C001',
      settings: { pulls: false },
    });

    const message = new UpdatedSettings({ subscription, resource: repository });
    expect(message.toJSON().attachments[0].text).not.toMatch(/pulls/);
    expect(message.toJSON()).toMatchSnapshot();
  });

  test('shows array elements', async () => {
    const subscription = new Subscription({
      channelId: 'C001',
      settings: { label: ['todo', 'help wanted'] },
    });

    const message = new UpdatedSettings({ subscription, resource: repository });

    const labelString = "label: ['todo', 'help wanted']";
    expect(message.toJSON().attachments[0].text).toMatch(labelString);

    expect(message.toJSON()).toMatchSnapshot();
  });

  test('hides empty array', async () => {
    const subscription = new Subscription({
      channelId: 'C001',
      settings: { label: [] },
    });

    const message = new UpdatedSettings({ subscription, resource: repository });
    expect(message.toJSON().attachments[0].text).not.toMatch(/label/);
    expect(message.toJSON()).toMatchSnapshot();
  });
});
