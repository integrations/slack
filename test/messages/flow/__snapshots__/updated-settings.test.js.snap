// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatedSettings message default settings 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "footer": "<https://github.com/integrations/slack#configuration|Learn More>",
      "mrkdwn_in": Array [
        "text",
        "footer",
      ],
      "text": "This channel will get notifications from <https://github.com/bkeepers/dotenv|bkeepers/dotenv> for: 
\`issues\`, \`pulls\`, \`deployments\`, \`statuses\`, \`public\`, \`commits\`, \`releases\`",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`UpdatedSettings message hides empty array 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "footer": "<https://github.com/integrations/slack#configuration|Learn More>",
      "mrkdwn_in": Array [
        "text",
        "footer",
      ],
      "text": "This channel will get notifications from <https://github.com/bkeepers/dotenv|bkeepers/dotenv> for: 
\`issues\`, \`pulls\`, \`deployments\`, \`statuses\`, \`public\`, \`commits\`, \`releases\`",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`UpdatedSettings message hides overwritten settings 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "footer": "<https://github.com/integrations/slack#configuration|Learn More>",
      "mrkdwn_in": Array [
        "text",
        "footer",
      ],
      "text": "This channel will get notifications from <https://github.com/bkeepers/dotenv|bkeepers/dotenv> for: 
\`issues\`, \`deployments\`, \`statuses\`, \`public\`, \`commits\`, \`releases\`",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`UpdatedSettings message shows array elements 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "footer": "<https://github.com/integrations/slack#configuration|Learn More>",
      "mrkdwn_in": Array [
        "text",
        "footer",
      ],
      "text": "This channel will get notifications from <https://github.com/bkeepers/dotenv|bkeepers/dotenv> for: 
\`issues\`, \`pulls\`, \`deployments\`, \`statuses\`, \`public\`, \`commits\`, \`releases\`, \`label: ['todo', 'help wanted']\`",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`UpdatedSettings message sub settings 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "footer": "<https://github.com/integrations/slack#configuration|Learn More>",
      "mrkdwn_in": Array [
        "text",
        "footer",
      ],
      "text": "This channel will get notifications from <https://github.com/bkeepers/dotenv|bkeepers/dotenv> for: 
\`issues\`, \`pulls\`, \`deployments\`, \`statuses\`, \`public\`, \`commits:all\`, \`releases\`",
    },
  ],
  "response_type": "in_channel",
}
`;
